// @apps-in-toss/framework/dist/index.cjs
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __reExport = (target, mod, secondTarget) => (__copyProps(target, mod, "default"), secondTarget && __copyProps(secondTarget, mod, "default"));
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  Analytics: () => Analytics2,
  AppsInToss: () => AppsInToss,
  INTERNAL__onVisibilityChangedByTransparentServiceWeb: () => INTERNAL__onVisibilityChangedByTransparentServiceWeb,
  OverlayProvider: () => import_private10.OverlayProvider,
  WebView: () => WebView,
  env: () => env,
  useCreateUserAgent: () => useCreateUserAgent,
  useGeolocation: () => useGeolocation,
  useOverlay: () => import_private10.useOverlay,
  useTopNavigation: () => import_private10.useTopNavigation,
  useWaitForReturnNavigator: () => useWaitForReturnNavigator
});
module.exports = __toCommonJS(src_exports);
var import_analytics2 = require("@apps-in-toss/analytics");

// src/core/registerApp.tsx
var import_analytics = require("@apps-in-toss/analytics");
var import_native_modules13 = require("@apps-in-toss/native-modules");
var import_react_native20 = require("@granite-js/react-native");
var import_tds_react_native10 = require("@toss/tds-react-native");
var import_react_native21 = require("react-native");

// src/core/components/AppEvent.tsx
var import_native_modules2 = require("@apps-in-toss/native-modules");
var import_react_native4 = require("@granite-js/react-native");
var import_react3 = require("react");

// src/env.ts
var env = {
  getDeploymentId: () => __DEV__ ? "local" : global.__appsInToss?.deploymentId
};

// src/hooks/useCaptureExitLog.ts
var import_native_modules = require("@apps-in-toss/native-modules");
var import_react_native3 = require("@granite-js/react-native");
var import_react2 = require("react");

// src/core/hooks/useReferrer.ts
var import_react_native = require("@granite-js/react-native");
var import_react = require("react");
function useReferrer() {
  return (0, import_react.useMemo)(() => {
    try {
      return new URL((0, import_react_native.getSchemeUri)()).searchParams.get("referrer");
    } catch {
      return null;
    }
  }, []);
}

// src/utils/isPrivateScheme.ts
var import_react_native2 = require("@granite-js/react-native");
function isPrivateScheme() {
  try {
    return new URL((0, import_react_native2.getSchemeUri)()).protocol === "intoss-private:";
  } catch {
    return false;
  }
}

// src/hooks/useCaptureExitLog.ts
var EXIT_IMPRESSION_LOG_NAME = "appsintoss_app_visit__common_module::impression__stay_time";
var EXIT_IMPRESSION_SCHEMA_ID = 1631628;
function useCaptureExitLog() {
  const referrer = useReferrer();
  const visible = (0, import_react_native3.useVisibility)();
  const enterTime = (0, import_react2.useRef)(void 0);
  (0, import_react2.useEffect)(() => {
    if (visible === true) {
      enterTime.current = Date.now();
    }
  }, [visible]);
  const captureExitLog = (0, import_react2.useCallback)(
    (exitTime) => {
      if (enterTime.current == null) {
        return;
      }
      const stayTime = Math.floor(exitTime - enterTime.current);
      import_native_modules.INTERNAL__module.tossCoreEventLog({
        log_name: EXIT_IMPRESSION_LOG_NAME,
        log_type: "event",
        params: {
          schema_id: EXIT_IMPRESSION_SCHEMA_ID,
          event_type: "impression",
          referrer,
          deployment_id: env.getDeploymentId(),
          app_name: import_react_native3.Granite.appName,
          is_private: isPrivateScheme(),
          stay_time: stayTime.toString(),
          exit_time: exitTime.toString()
        }
      });
      enterTime.current = void 0;
    },
    [referrer]
  );
  return { captureExitLog };
}

// src/core/components/AppEvent.tsx
var ENTRY_APP_EVENT_SCHEMA_ID = 1562181;
function EntryAppEvent() {
  const referrer = useReferrer() ?? "";
  (0, import_react3.useEffect)(() => {
    import_native_modules2.INTERNAL__module.tossCoreEventLog({
      log_name: "appsintoss_app_visit::impression__enter_appsintoss",
      log_type: "info",
      params: {
        is_transform: true,
        schema_id: ENTRY_APP_EVENT_SCHEMA_ID,
        referrer,
        deployment_id: env.getDeploymentId(),
        app_name: import_react_native4.Granite.appName,
        is_private: isPrivateScheme()
      }
    });
  }, [referrer]);
  return null;
}
function SystemAppEvent({ ...initialProps }) {
  (0, import_react3.useEffect)(() => {
    import_native_modules2.INTERNAL__module.tossCoreEventLog({
      log_name: "AppsInTossInitialProps",
      log_type: "debug",
      params: {
        ...initialProps,
        schemeUri: (0, import_react_native4.getSchemeUri)(),
        deployment_id: env.getDeploymentId(),
        app_name: import_react_native4.Granite.appName,
        is_private: isPrivateScheme()
      }
    });
  }, [initialProps]);
  return null;
}
function StayTimeAppEvent() {
  const visible = (0, import_react_native4.useVisibility)();
  const { captureExitLog } = useCaptureExitLog();
  (0, import_react3.useEffect)(() => {
    if (visible === false) {
      captureExitLog(Date.now());
    }
  }, [visible, captureExitLog]);
  return null;
}
var AppEvent = {
  Entry: EntryAppEvent,
  System: SystemAppEvent,
  StayTime: StayTimeAppEvent
};

// src/core/components/AppUpdate.tsx
var import_react_native5 = require("@granite-js/react-native");
var import_react4 = require("react");
var import_jsx_runtime = require("react/jsx-runtime");
var UPDATE_SCHEME = "servicetoss://update/forced/j?updateType=required";
function AppUpdate() {
  (0, import_react4.useEffect)(() => {
    (0, import_react_native5.openURL)(`supertoss://closePage?url=${UPDATE_SCHEME}`);
  }, []);
  return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_jsx_runtime.Fragment, {});
}

// src/core/hooks/useAppsInTossBridge.ts
var import_native_modules3 = require("@apps-in-toss/native-modules");
var import_tds_react_native = require("@toss/tds-react-native");
var import_react5 = require("react");

// src/core/utils/getAppsInTossGlobals.ts
function getAppsInTossGlobals() {
  if (global.__appsInToss == null) {
    throw new Error("invalid apps-in-toss globals");
  }
  return global.__appsInToss;
}

// src/core/utils/toIcon.ts
function toIcon(source) {
  return source.startsWith("http") ? { source: { uri: source } } : { name: source };
}

// src/core/hooks/useAppsInTossBridge.ts
function useAppsInTossBridge() {
  const controller = (0, import_tds_react_native.useBridge)();
  const appsInTossGlobals = getAppsInTossGlobals();
  (0, import_react5.useEffect)(() => {
    const isGameApp = appsInTossGlobals.webViewType === "game" || appsInTossGlobals.appType === "game";
    controller.open({
      serviceName: appsInTossGlobals.brandDisplayName,
      icon: toIcon(appsInTossGlobals.brandIcon),
      color: appsInTossGlobals.brandPrimaryColor,
      colorMode: isGameApp ? "inverted" : appsInTossGlobals.brandBridgeColorMode,
      onExited: () => {
        import_native_modules3.appsInTossEvent.emit("entryMessageExited", void 0);
      }
    });
  }, []);
}

// src/components/RNAppContainer.tsx
var import_native_modules12 = require("@apps-in-toss/native-modules");
var import_react14 = require("react");
var import_react_native19 = require("react-native");

// src/components/GameProfile.tsx
var import_native_modules6 = require("@apps-in-toss/native-modules");
var import_tds_react_native4 = require("@toss/tds-react-native");
var import_react7 = require("react");
var import_react_native9 = require("react-native");

// src/constant/game-center.ts
var GAME_PROFILE_WEBVIEW_URL = "servicetoss://game-center/profile";
var GAME_CENTER_MIN_VERSION = {
  android: "5.221.0",
  ios: "5.221.0"
};

// src/hooks/useGameCenterProfile.ts
var import_native_modules5 = require("@apps-in-toss/native-modules");
var import_react_native8 = require("@granite-js/react-native");
var import_tds_react_native3 = require("@toss/tds-react-native");
var import_es_hangul = require("es-hangul");
var import_react6 = require("react");

// src/components/GameProfileToast.tsx
var import_tds_react_native2 = require("@toss/tds-react-native");
var import_private = require("@toss/tds-react-native/private");
var import_jsx_runtime2 = require("react/jsx-runtime");
var useGameProfileToast = () => {
  const overlay = (0, import_private.useOverlay)();
  const openGameProfileToast = (nickname, profileImageUri) => {
    return new Promise((resolve) => {
      overlay.open(({ isOpen, close, exit }) => {
        return /* @__PURE__ */ (0, import_jsx_runtime2.jsx)(import_private.ColorPreferenceProvider, { colorPreference: "dark", children: /* @__PURE__ */ (0, import_jsx_runtime2.jsx)(import_private.AdaptiveColorProvider, { children: /* @__PURE__ */ (0, import_jsx_runtime2.jsx)(
          import_tds_react_native2.Toast,
          {
            open: isOpen,
            onClose: () => {
              resolve();
              close();
            },
            onExited: exit,
            position: "top",
            text: `${nickname}\uB2D8 \uBC18\uAC00\uC6CC\uC694!`,
            icon: /* @__PURE__ */ (0, import_jsx_runtime2.jsx)(
              import_tds_react_native2.Asset.Image,
              {
                style: { borderRadius: 64, overflow: "hidden" },
                frameShape: import_tds_react_native2.Asset.frameShape.CleanW32,
                source: { uri: profileImageUri }
              }
            )
          }
        ) }) });
      });
    });
  };
  return { openGameProfileToast };
};

// src/utils/error.ts
var DEFAULT_ERROR = {
  title: "\uC7A0\uC2DC \uD6C4 \uB2E4\uC2DC \uC2DC\uB3C4\uD574\uC8FC\uC138\uC694",
  description: "\uBB38\uC81C\uAC00 \uACC4\uC18D\uB418\uBA74 \uD1A0\uC2A4 \uACE0\uAC1D\uC13C\uD130(1599-4905)\uB85C \uBB38\uC758\uD574\uC8FC\uC138\uC694."
};

// src/utils/market.ts
var import_react_native6 = require("react-native");
var PLAYSTORE_LINK = "https://play.google.com/store/apps/details?id=viva.republica.toss";
var APPSTORE_LINK = "https://itunes.apple.com/app/id839333328";
var getMarketLink = () => {
  return import_react_native6.Platform.OS === "android" ? PLAYSTORE_LINK : APPSTORE_LINK;
};

// src/utils/openTransparentWebView.ts
var import_react_native7 = require("@granite-js/react-native");

// src/private.ts
var import_native_modules4 = require("@apps-in-toss/native-modules");
var INTERNAL__onVisibilityChangedByTransparentServiceWeb = import_native_modules4.onVisibilityChangedByTransparentServiceWeb;

// src/utils/openTransparentWebView.ts
var openTransparentWebView = ({
  webUrl,
  cleanupWhenDismissed = true,
  onEvent,
  onError,
  callbackId = "fn",
  params
}) => {
  const url = new URL("supertoss://transparent-service-web");
  url.searchParams.set("url", webUrl);
  url.searchParams.set("onVisibilityChangeCallback", callbackId);
  Object.entries(params ?? {}).forEach(([key, value]) => {
    url.searchParams.set(key, value);
  });
  const cleanup = INTERNAL__onVisibilityChangedByTransparentServiceWeb({
    options: { callbackId },
    onError: (error) => {
      onError(error);
      cleanup();
    },
    onEvent: (value) => {
      onEvent(value);
      if (cleanupWhenDismissed && value === true) {
        cleanup();
      }
    }
  });
  (0, import_react_native7.openURL)(url.toString());
};

// src/hooks/useGameCenterProfile.ts
var useGameCenterProfile = (isReadyForProfileUI) => {
  const [profileData, setProfileData] = (0, import_react6.useState)(void 0);
  const [isProfileDataLoading, setIsProfileDataLoading] = (0, import_react6.useState)(true);
  const [isProfileDataRefetching, setIsProfileDataRefetching] = (0, import_react6.useState)(false);
  const shouldShowLoadingOverlay = isProfileDataLoading && isReadyForProfileUI;
  const shouldShowProfileNotFoundOverlay = profileData?.statusCode === "PROFILE_NOT_FOUND" && isReadyForProfileUI && !isProfileDataRefetching;
  const canShowBottomSheetOrToast = !isProfileDataLoading && isReadyForProfileUI;
  const [isWebviewLoading, setIsWebviewLoading] = (0, import_react6.useState)(false);
  const isCompletedProfileFlow = (0, import_react6.useRef)(false);
  const { openAlert, openConfirm } = (0, import_tds_react_native3.useDialog)();
  const { openGameProfileToast } = useGameProfileToast();
  const openErrorAlert = (0, import_react6.useCallback)(async () => {
    await openAlert({
      title: DEFAULT_ERROR.title,
      description: DEFAULT_ERROR.description
    });
    (0, import_react_native8.closeView)();
  }, [openAlert]);
  const openProfileWebview = (0, import_react6.useCallback)(() => {
    if (isWebviewLoading) {
      return;
    }
    setIsWebviewLoading(true);
    openTransparentWebView({
      webUrl: `${GAME_PROFILE_WEBVIEW_URL}?appName=${getAppName()}&referrer=appsintoss.${getAppName()}`,
      onEvent: async (isClosedTransparentWebView) => {
        if (isClosedTransparentWebView) {
          try {
            setIsWebviewLoading(false);
            setIsProfileDataRefetching(true);
            const data = await (0, import_native_modules5.getGameCenterGameProfile)();
            setProfileData(data);
            setIsProfileDataRefetching(false);
            if (data?.statusCode === "SUCCESS") {
              openGameProfileToast(data.nickname, data.profileImageUri);
            }
          } catch (_) {
            setIsProfileDataRefetching(false);
            openErrorAlert();
          }
        }
      },
      onError: () => {
        openErrorAlert();
      }
    });
  }, [isWebviewLoading, openGameProfileToast, openErrorAlert]);
  const updateAppToSupportedMinVersion = (0, import_react6.useCallback)(async () => {
    const upddateConfirmDialogLabel = {
      title: `${(0, import_es_hangul.josa)(getAppsInTossGlobals().brandDisplayName, "\uC744/\uB97C")} \uD558\uB824\uBA74 
\uC571\uC744 \uC5C5\uB370\uC774\uD2B8\uD574\uC8FC\uC138\uC694`,
      leftButton: "\uB2EB\uAE30",
      rightButton: "\uC5C5\uB370\uC774\uD2B8\uD558\uAE30"
    };
    const isConfirmed = await openConfirm({
      title: upddateConfirmDialogLabel.title,
      leftButton: upddateConfirmDialogLabel.leftButton,
      rightButton: upddateConfirmDialogLabel.rightButton,
      closeOnDimmerClick: true
    });
    if (!isConfirmed) {
      (0, import_react_native8.closeView)();
      return;
    }
    const STORE_SCHEME = getMarketLink();
    (0, import_react_native8.openURL)(`supertoss://web?url=${STORE_SCHEME}&external=browser`);
  }, [openConfirm]);
  return {
    profileData,
    isProfileDataLoading,
    isProfileDataRefetching,
    shouldShowLoadingOverlay,
    shouldShowProfileNotFoundOverlay,
    canShowBottomSheetOrToast,
    isCompletedProfileFlow,
    updateAppToSupportedMinVersion,
    setIsProfileDataLoading,
    openProfileWebview,
    setProfileData,
    openErrorAlert,
    openGameProfileToast
  };
};

// src/utils/zIndex.ts
var Z_INDEX = {
  /* 게임 프로필을 위한 overlay
   */
  PROFILE_OVERLAY: 9998,
  // 게임을 종료할 수 있는 X 버튼
  CLOSE_BUTTON: 9999
};

// src/components/GameProfile.tsx
var import_jsx_runtime3 = require("react/jsx-runtime");
var GameProfile = ({ children, isReadyForProfileUI }) => {
  const {
    profileData,
    isProfileDataRefetching,
    shouldShowLoadingOverlay,
    shouldShowProfileNotFoundOverlay,
    canShowBottomSheetOrToast,
    isCompletedProfileFlow,
    openProfileWebview,
    updateAppToSupportedMinVersion,
    setIsProfileDataLoading,
    setProfileData,
    openErrorAlert,
    openGameProfileToast
  } = useGameCenterProfile(isReadyForProfileUI);
  (0, import_react7.useEffect)(() => {
    try {
      const getProfileData = async () => {
        const data = await (0, import_native_modules6.getGameCenterGameProfile)();
        setProfileData(data);
        setIsProfileDataLoading(false);
      };
      getProfileData();
    } catch (_) {
      openErrorAlert();
      setIsProfileDataLoading(false);
    }
  }, []);
  (0, import_react7.useEffect)(() => {
    const handleGameProfileFlow = async () => {
      if (!canShowBottomSheetOrToast) {
        return;
      }
      if (isCompletedProfileFlow.current) {
        return;
      }
      isCompletedProfileFlow.current = true;
      if (!(0, import_native_modules6.isMinVersionSupported)(GAME_CENTER_MIN_VERSION)) {
        updateAppToSupportedMinVersion();
        return;
      }
      if (profileData?.statusCode === "SUCCESS") {
        openGameProfileToast(profileData.nickname, profileData.profileImageUri);
        return;
      }
      if (profileData?.statusCode === "PROFILE_NOT_FOUND") {
        openProfileWebview();
      }
    };
    handleGameProfileFlow();
  }, [
    canShowBottomSheetOrToast,
    isCompletedProfileFlow,
    openGameProfileToast,
    openProfileWebview,
    profileData,
    updateAppToSupportedMinVersion
  ]);
  if (!(0, import_native_modules6.isMinVersionSupported)(GAME_CENTER_MIN_VERSION)) {
    return /* @__PURE__ */ (0, import_jsx_runtime3.jsxs)(import_jsx_runtime3.Fragment, { children: [
      /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(import_react_native9.View, { style: { flex: 1, position: "relative" }, children }),
      /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(
        import_react_native9.Pressable,
        {
          style: {
            ...overlayStyle
          },
          onPress: () => {
            updateAppToSupportedMinVersion();
          }
        }
      )
    ] });
  }
  if (shouldShowLoadingOverlay || isProfileDataRefetching) {
    return /* @__PURE__ */ (0, import_jsx_runtime3.jsxs)(import_jsx_runtime3.Fragment, { children: [
      /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(import_react_native9.View, { style: { flex: 1, position: "relative" }, children }),
      /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(
        import_react_native9.View,
        {
          style: {
            ...overlayStyle,
            justifyContent: "center",
            alignItems: "center",
            backgroundColor: "rgba(0, 0, 0, 0.2)"
          },
          children: /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(import_tds_react_native4.Loader, { size: "large", type: "light" })
        }
      )
    ] });
  }
  if (shouldShowProfileNotFoundOverlay) {
    return /* @__PURE__ */ (0, import_jsx_runtime3.jsxs)(import_jsx_runtime3.Fragment, { children: [
      /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(import_react_native9.View, { style: { flex: 1, position: "relative" }, children }),
      shouldShowProfileNotFoundOverlay && /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(
        import_react_native9.Pressable,
        {
          style: {
            ...overlayStyle
          },
          onPress: () => {
            openProfileWebview();
          }
        }
      )
    ] });
  }
  return /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(import_jsx_runtime3.Fragment, { children: /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(import_react_native9.View, { style: { flex: 1, position: "relative" }, children }) });
};
var overlayStyle = {
  position: "absolute",
  top: 0,
  left: 0,
  right: 0,
  bottom: 0,
  zIndex: Z_INDEX.PROFILE_OVERLAY
};

// src/components/NavigationBar/RNNavigationBar/Default.tsx
var import_private5 = require("@toss/tds-react-native/private");

// src/components/NavigationBar/RNNavigationBar/hooks/useHardwareBackPress.ts
var import_react8 = require("react");
var import_react_native10 = require("react-native");
function useHardwareBackPress(handler) {
  (0, import_react8.useEffect)(() => {
    const handleBackPress = () => {
      handler();
      return true;
    };
    import_react_native10.BackHandler.addEventListener("hardwareBackPress", handleBackPress);
    return () => import_react_native10.BackHandler.removeEventListener("hardwareBackPress", handleBackPress);
  }, [handler]);
}

// src/components/NavigationBar/RNNavigationBar/hooks/useNavigationEvent.ts
var import_react_native12 = require("@granite-js/react-native");
var import_react10 = require("react");

// src/components/NavigationBar/RNNavigationBar/hooks/useCloseConfirm.ts
var import_tds_react_native5 = require("@toss/tds-react-native");
var import_es_hangul2 = require("es-hangul");
var import_react9 = require("react");
function useCloseConfirm() {
  const { brandDisplayName } = getAppsInTossGlobals();
  const { openConfirm } = (0, import_tds_react_native5.useDialog)();
  return (0, import_react9.useCallback)(async ({ onEntered }) => {
    return await openConfirm({
      title: `${(0, import_es_hangul2.josa)(brandDisplayName, "\uC744/\uB97C")} \uC885\uB8CC\uD560\uAE4C\uC694?`,
      leftButton: "\uCDE8\uC18C",
      rightButton: "\uC885\uB8CC\uD558\uAE30",
      closeOnDimmerClick: true,
      onEntered
    });
  }, [brandDisplayName, openConfirm]);
}

// src/components/NavigationBar/common/useNavigationBarLogging.tsx
var import_native_modules7 = require("@apps-in-toss/native-modules");
var import_react_native11 = require("@granite-js/react-native");
var NAVI_BAR_IMPRESSION_SCHEMA_ID = 1596837;
var NAVI_BAR_IMPRESSION_LOG_NAME = "appsintoss_app_visit__common_module::impression__navigation_bar";
var CLOSE_POPUP_SHOW_SCHEMA_ID = 1644490;
var CLOSE_POPUP_SHOW_LOG_NAME = "appsintoss_app_visit__common_module::popup__close_app";
var CLOSE_BUTTON_CLICK_SCHEMA_ID = 1596831;
var CLOSE_BUTTON_CLICK_LOG_NAME = "appsintoss_app_visit__common_module::click__icon_close";
var CLOSE_POPUP_CTA_CLICK_SCHEMA_ID = 1644492;
var CLOSE_POPUP_CTA_CLICK_LOG_NAME = "appsintoss_app_visit__common_module::popup__close_app::click__cta";
var HOME_BUTTON_CLICK_SCHEMA_ID = 1596839;
var HOME_BUTTON_CLICK_LOG_NAME = "appsintoss_app_visit__common_module::click__icon_home";
function useNavigationBarLogging() {
  const referrer = useReferrer();
  const baseParams = {
    referrer,
    app_name: import_react_native11.Granite.appName
  };
  const logNavBarImpression = (naviBarConfig) => {
    import_native_modules7.INTERNAL__module.tossCoreEventLog({
      log_name: NAVI_BAR_IMPRESSION_LOG_NAME,
      log_type: "event",
      params: {
        ...naviBarConfig,
        ...baseParams,
        event_type: "impression",
        schema_id: NAVI_BAR_IMPRESSION_SCHEMA_ID
      }
    });
  };
  const logHomeButtonClick = () => {
    import_native_modules7.INTERNAL__module.tossCoreEventLog({
      log_name: HOME_BUTTON_CLICK_LOG_NAME,
      log_type: "event",
      params: {
        ...baseParams,
        event_type: "click",
        schema_id: HOME_BUTTON_CLICK_SCHEMA_ID
      }
    });
  };
  const logCloseButtonClick = () => {
    import_native_modules7.INTERNAL__module.tossCoreEventLog({
      log_name: CLOSE_BUTTON_CLICK_LOG_NAME,
      log_type: "event",
      params: {
        ...baseParams,
        event_type: "click",
        schema_id: CLOSE_BUTTON_CLICK_SCHEMA_ID
      }
    });
  };
  const logClosePopupShow = () => {
    import_native_modules7.INTERNAL__module.tossCoreEventLog({
      log_name: CLOSE_POPUP_SHOW_LOG_NAME,
      log_type: "popup",
      params: {
        ...baseParams,
        schema_id: CLOSE_POPUP_SHOW_SCHEMA_ID
      }
    });
  };
  const logClosePopupCtaClick = (confirm) => {
    import_native_modules7.INTERNAL__module.tossCoreEventLog({
      log_name: CLOSE_POPUP_CTA_CLICK_LOG_NAME,
      log_type: "event",
      params: {
        ...baseParams,
        close_yn: confirm ? "Y" : "N",
        schema_id: CLOSE_POPUP_CTA_CLICK_SCHEMA_ID,
        event_type: "click"
      }
    });
  };
  return {
    navBarImpression: logNavBarImpression,
    closePopupShow: logClosePopupShow,
    closePopupCtaClick: logClosePopupCtaClick,
    closeButtonClick: logCloseButtonClick,
    homeButtonClick: logHomeButtonClick
  };
}

// src/components/NavigationBar/RNNavigationBar/hooks/useNavigationEvent.ts
function useNavigationEvent() {
  const backEventContext = (0, import_react_native12.useBackEventContext)();
  const logging = useNavigationBarLogging();
  const navigation = (0, import_react_native12.useNavigation)();
  const closeConfirm = useCloseConfirm();
  const { captureExitLog } = useCaptureExitLog();
  return (0, import_react10.useMemo)(() => {
    const close = async () => {
      const hasConfirmed = await closeConfirm({
        onEntered: logging.closePopupShow
      });
      logging.closePopupCtaClick(hasConfirmed);
      if (hasConfirmed) {
        captureExitLog(Date.now());
        (0, import_react_native12.closeView)();
      }
    };
    return {
      handleBack: () => {
        if (backEventContext.hasBackEvent) {
          backEventContext.onBack();
        } else if (navigation.canGoBack()) {
          navigation.goBack();
        } else {
          close();
        }
      },
      handleHomeButtonClick: () => {
        logging.homeButtonClick();
        navigation.navigate("/");
      },
      handleCloseButtonClick: () => {
        logging.closeButtonClick();
        close();
      }
    };
  }, [backEventContext, navigation, closeConfirm, logging, captureExitLog]);
}

// src/core/hooks/useMoreButtonBottomSheet/index.tsx
var import_native_modules11 = require("@apps-in-toss/native-modules");
var import_react_native17 = require("@granite-js/react-native");
var import_tds_react_native8 = require("@toss/tds-react-native");
var import_private4 = require("@toss/tds-react-native/private");
var import_react12 = require("react");

// src/core/hooks/useMoreButtonBottomSheet/AppShareListMenu.tsx
var import_native_modules9 = require("@apps-in-toss/native-modules");
var import_react_native14 = require("@granite-js/react-native");
var import_tds_react_native6 = require("@toss/tds-react-native");
var import_private3 = require("@toss/tds-react-native/private");
var import_react_native15 = require("react-native");

// src/core/hooks/useMoreButtonBottomSheet/useMoreButtonBottomSheetLogging.tsx
var import_native_modules8 = require("@apps-in-toss/native-modules");
var import_react_native13 = require("@granite-js/react-native");
var BOTTOM_SHEET_SCHEMA_ID = 1596825;
var BOTTOM_SHEET_LOG_NAME = "appsintoss_app_visit__common_module::bottomsheet__more";
var BOTTOM_SHEET_OPEN_SCHEMA_ID = 1596829;
var BOTTOM_SHEET_OPEN_LOG_NAME = "appsintoss_app_visit__common_module::click__icon_more";
var BOTTOM_SHEET_CLOSE_CLICK_SCHEMA_ID = 1596843;
var BOTTOM_SHEET_CLOSE_CLICK_LOG_NAME = "appsintoss_app_visit__common_module::bottomsheet__more::click__close";
var BOTTOM_SHEET_MENU_CLICK_SCHEMA_ID = 1596841;
var BOTTOM_SHEET_MENU_CLICK_LOG_NAME = "appsintoss_app_visit__common_module::bottomsheet__more::click__menu";
function useMoreButtonBottomSheetLogging() {
  const referrer = useReferrer();
  const baseParams = {
    referrer,
    app_name: import_react_native13.Granite.appName
  };
  const logBottomSheetShow = () => {
    import_native_modules8.INTERNAL__module.tossCoreEventLog({
      log_name: BOTTOM_SHEET_LOG_NAME,
      log_type: "popup",
      params: {
        ...baseParams,
        schema_id: BOTTOM_SHEET_SCHEMA_ID
      }
    });
  };
  const logBottomSheetOpen = () => {
    import_native_modules8.INTERNAL__module.tossCoreEventLog({
      log_name: BOTTOM_SHEET_OPEN_LOG_NAME,
      log_type: "event",
      params: {
        ...baseParams,
        schema_id: BOTTOM_SHEET_OPEN_SCHEMA_ID,
        event_type: "click"
      }
    });
  };
  const logBottomSheetCloseClick = () => {
    import_native_modules8.INTERNAL__module.tossCoreEventLog({
      log_name: BOTTOM_SHEET_CLOSE_CLICK_LOG_NAME,
      log_type: "event",
      params: {
        ...baseParams,
        schema_id: BOTTOM_SHEET_CLOSE_CLICK_SCHEMA_ID,
        event_type: "click"
      }
    });
  };
  const logBottomSheetMenuClick = ({ title }) => {
    import_native_modules8.INTERNAL__module.tossCoreEventLog({
      log_name: BOTTOM_SHEET_MENU_CLICK_LOG_NAME,
      log_type: "event",
      params: {
        ...baseParams,
        schema_id: BOTTOM_SHEET_MENU_CLICK_SCHEMA_ID,
        event_type: "click",
        item_title: title
      }
    });
  };
  return {
    show: logBottomSheetShow,
    open: logBottomSheetOpen,
    close: logBottomSheetCloseClick,
    menuClick: logBottomSheetMenuClick
  };
}

// src/core/utils/url.ts
function addParamsToUrl(url, params) {
  const urlObj = new URL(url);
  for (const [key, value] of Object.entries(params)) {
    urlObj.searchParams.set(key, value);
  }
  return urlObj.toString();
}

// src/core/hooks/useMoreButtonBottomSheet/AppShareListMenu.tsx
var import_jsx_runtime4 = require("react/jsx-runtime");
var SHARE_SCHEME_REFERRER = "appsintoss.common_module_share";
var APP_SHARE_MENU_INFO = {
  contactItemName: "\uACF5\uC720\uD558\uAE30",
  contactIconUrl: "https://static.toss.im/icons/png/4x/icon-share-dots-mono.png"
};
function AppShareListMenu() {
  const globals = getAppsInTossGlobals();
  const adaptive = (0, import_private3.useAdaptive)();
  const logging = useMoreButtonBottomSheetLogging();
  const initialScheme = (0, import_react_native14.getSchemeUri)();
  const isSandbox = (0, import_native_modules9.getOperationalEnvironment)() === "sandbox";
  const { openConfirm } = (0, import_tds_react_native6.useDialog)();
  const schemeForShare = addParamsToUrl(initialScheme, {
    referrer: SHARE_SCHEME_REFERRER
  });
  return /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(
    import_tds_react_native6.ListRow,
    {
      left: /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(
        import_tds_react_native6.ListRow.Icon,
        {
          color: globals.brandPrimaryColor,
          source: { uri: APP_SHARE_MENU_INFO.contactIconUrl },
          type: "background"
        }
      ),
      contents: /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(
        import_tds_react_native6.ListRow.Texts,
        {
          type: "1RowTypeA",
          top: APP_SHARE_MENU_INFO.contactItemName,
          topProps: { color: adaptive.grey700 }
        }
      ),
      verticalPadding: "extraSmall",
      onPress: () => {
        if (isSandbox) {
          openConfirm({
            title: "\uACF5\uC720\uD558\uAE30 \uAE30\uB2A5 \uBBF8\uC9C0\uC6D0",
            description: "\uC0CC\uB4DC\uBC15\uC2A4 \uD658\uACBD\uC5D0\uC11C\uB294 \uC0AC\uC6A9\uD560 \uC218 \uC5C6\uC5B4\uC694. \uCF58\uC194\uC744 \uD1B5\uD574 \uD1A0\uC2A4\uC571\uC5D0\uC11C \uD14C\uC2A4\uD2B8\uD574 \uC8FC\uC138\uC694.",
            rightButton: "\uD655\uC778",
            closeOnDimmerClick: true
          });
          return;
        }
        logging.menuClick({ title: APP_SHARE_MENU_INFO.contactItemName });
        import_react_native15.NativeModules.AppsInTossModule.shareWithScheme({
          params: {
            schemeURL: schemeForShare
          }
        });
      }
    }
  );
}

// src/hooks/useAppUpdateDialog.tsx
var import_native_modules10 = require("@apps-in-toss/native-modules");
var import_react_native16 = require("@granite-js/react-native");
var import_tds_react_native7 = require("@toss/tds-react-native");
var import_react11 = require("react");
function useAppUpdateDialog() {
  const { openConfirm } = (0, import_tds_react_native7.useDialog)();
  const logging = useAppUpdateDialogLogging();
  const openAppUpdateDialog = (0, import_react11.useCallback)(
    async ({
      title,
      description,
      leftButton = "\uB2EB\uAE30",
      rightButton = "\uC5C5\uB370\uC774\uD2B8\uD558\uAE30"
    }) => {
      logging.show();
      const isConfirmed = await openConfirm({
        title,
        description,
        leftButton,
        rightButton,
        closeOnDimmerClick: true
      });
      if (!isConfirmed) {
        logging.close();
        return;
      }
      logging.update();
      const STORE_SCHEME = getMarketLink();
      (0, import_react_native16.openURL)(`supertoss://web?url=${STORE_SCHEME}&external=browser`);
    },
    [logging, openConfirm]
  );
  return {
    open: openAppUpdateDialog
  };
}
var UPDATE_DIALOG_SCHEMA_ID = 1634992;
var UPDATE_DIALOG_LOG_NAME = "appsintoss_app_visit__common_module::bottomsheet__app_update";
var UPDATE_DIALOG_CTA_CLICK_SCHEMA_ID = 1634996;
var UPDATE_DIALOG_CTA_CLICK_LOG_NAME = "appsintoss_app_visit__common_module::bottomsheet__app_update::click__cta";
function useAppUpdateDialogLogging() {
  const referrer = useReferrer();
  const baseParams = {
    referrer,
    app_name: import_react_native16.Granite.appName
  };
  const logUpdateClick = () => {
    import_native_modules10.INTERNAL__module.tossCoreEventLog({
      log_name: UPDATE_DIALOG_CTA_CLICK_LOG_NAME,
      log_type: "event",
      params: {
        ...baseParams,
        schema_id: UPDATE_DIALOG_CTA_CLICK_SCHEMA_ID,
        event_type: "click",
        button_type: "update"
      }
    });
  };
  const logCloseClick = () => {
    import_native_modules10.INTERNAL__module.tossCoreEventLog({
      log_name: UPDATE_DIALOG_CTA_CLICK_LOG_NAME,
      log_type: "event",
      params: {
        ...baseParams,
        schema_id: UPDATE_DIALOG_CTA_CLICK_SCHEMA_ID,
        event_type: "click",
        button_type: "close"
      }
    });
  };
  const logDialogShow = () => {
    import_native_modules10.INTERNAL__module.tossCoreEventLog({
      log_name: UPDATE_DIALOG_LOG_NAME,
      log_type: "popup",
      params: {
        ...baseParams,
        schema_id: UPDATE_DIALOG_SCHEMA_ID
      }
    });
  };
  return { update: logUpdateClick, close: logCloseClick, show: logDialogShow };
}

// src/core/utils/ensureValue.ts
function ensureValue(value, name) {
  if (value === void 0) {
    throw new Error(`${name} is required`);
  }
  return value;
}

// src/core/hooks/useMoreButtonBottomSheet/index.tsx
var import_jsx_runtime5 = require("react/jsx-runtime");
var APP_BRIDGE_METHOD_NAME = "getMiniAppsSupportContact";
var MIN_VERSION = {
  BOTTOM_SHEET: {
    android: "5.226.0",
    ios: "5.226.0"
  },
  SHARE_LIST_MENU: {
    android: "5.230.0",
    ios: "5.230.0"
  }
};
function useMoreButtonBottomSheet() {
  const globals = getAppsInTossGlobals();
  const adaptive = (0, import_private4.useAdaptive)();
  const [itemList, setItemList] = (0, import_react12.useState)([]);
  const appUpdateDialog = useAppUpdateDialog();
  const logging = useMoreButtonBottomSheetLogging();
  const overlay = (0, import_private4.useOverlay)();
  const title = ensureValue(globals.brandDisplayName, "displayName");
  const isBottomSheetSupported = (0, import_native_modules11.isMinVersionSupported)(MIN_VERSION.BOTTOM_SHEET);
  const isShareListMenuSupported = (0, import_native_modules11.isMinVersionSupported)(MIN_VERSION.SHARE_LIST_MENU);
  (0, import_react12.useEffect)(() => {
    if (!isBottomSheetSupported) {
      return;
    }
    import_native_modules11.INTERNAL__appBridgeHandler.invokeAppBridgeMethod(
      APP_BRIDGE_METHOD_NAME,
      {},
      {
        onSuccess: ({ items }) => setItemList(items),
        onError: (error) => console.error("\uBA54\uB274 \uBAA9\uB85D\uC744 \uAC00\uC838\uC624\uB294 \uB370 \uC2E4\uD328\uD588\uC5B4\uC694:", error)
      }
    );
  }, [isBottomSheetSupported]);
  const onClickHandler = async () => {
    logging.open();
    if (!isBottomSheetSupported) {
      await appUpdateDialog.open({
        title: `\uC774 \uAE30\uB2A5\uC744 \uC4F0\uB824\uBA74 \uC571 \uC5C5\uB370\uC774\uD2B8\uAC00 \uD544\uC694\uD574\uC694`,
        description: `\uBB38\uC758, \uAD8C\uD55C \uC124\uC815, \uC2E0\uACE0 \uB4F1 \uAE30\uB2A5\uC744 \uC4F8 \uC218 \uC788\uC5B4\uC694`
      });
      return;
    }
    overlay.open(({ isOpen, exit, close }) => {
      const handleClose = () => {
        logging.close();
        close();
      };
      return /* @__PURE__ */ (0, import_jsx_runtime5.jsx)(BottomSheetImpressionArea, { children: /* @__PURE__ */ (0, import_jsx_runtime5.jsx)(
        import_tds_react_native8.BottomSheet.Root,
        {
          header: /* @__PURE__ */ (0, import_jsx_runtime5.jsx)(
            import_tds_react_native8.ListHeader,
            {
              title: /* @__PURE__ */ (0, import_jsx_runtime5.jsx)(import_tds_react_native8.ListHeader.TitleParagraph, { color: adaptive.grey800, fontWeight: "bold", typography: "t5", children: title })
            }
          ),
          open: isOpen,
          cta: /* @__PURE__ */ (0, import_jsx_runtime5.jsx)(
            import_tds_react_native8.BottomSheet.CTA,
            {
              size: "large",
              type: "dark",
              style: "weak",
              onPress: () => {
                handleClose();
              },
              children: "\uB2EB\uAE30"
            }
          ),
          onClose: handleClose,
          onExited: exit,
          children: /* @__PURE__ */ (0, import_jsx_runtime5.jsxs)(import_tds_react_native8.List, { rowSeparator: "none", children: [
            itemList.map((item) => {
              return /* @__PURE__ */ (0, import_jsx_runtime5.jsx)(
                import_tds_react_native8.ListRow,
                {
                  left: /* @__PURE__ */ (0, import_jsx_runtime5.jsx)(
                    import_tds_react_native8.ListRow.Icon,
                    {
                      color: globals.brandPrimaryColor,
                      source: { uri: item.contactIconUrl },
                      type: "background"
                    }
                  ),
                  contents: /* @__PURE__ */ (0, import_jsx_runtime5.jsx)(
                    import_tds_react_native8.ListRow.Texts,
                    {
                      type: "1RowTypeA",
                      top: item.contactItemName,
                      topProps: { color: adaptive.grey700 }
                    }
                  ),
                  verticalPadding: "extraSmall",
                  onPress: () => {
                    logging.menuClick({ title: item.contactItemName });
                    (0, import_react_native17.openURL)(item.contactUri);
                  }
                },
                item.contactItemName
              );
            }),
            isShareListMenuSupported && /* @__PURE__ */ (0, import_jsx_runtime5.jsx)(AppShareListMenu, {})
          ] })
        }
      ) });
    });
  };
  return { open: onClickHandler };
}
function BottomSheetImpressionArea({ children }) {
  const logging = useMoreButtonBottomSheetLogging();
  (0, import_react12.useEffect)(() => {
    logging.show();
  }, [logging]);
  return /* @__PURE__ */ (0, import_jsx_runtime5.jsx)(import_jsx_runtime5.Fragment, { children });
}

// src/core/utils/safeParseNavigationBar.ts
function safeParseNavigationBar(navigationBar) {
  if (typeof navigationBar === "object") {
    return navigationBar;
  }
  try {
    return JSON.parse(navigationBar);
  } catch {
    return null;
  }
}

// src/components/NavigationBar/common/NavigationBarImpressionArea.tsx
var import_react13 = require("react");
var import_jsx_runtime6 = require("react/jsx-runtime");
function NavigationBarImpressionArea({
  children,
  withHomeButton
}) {
  const hasLogged = (0, import_react13.useRef)(false);
  const logging = useNavigationBarLogging();
  (0, import_react13.useEffect)(() => {
    if (hasLogged.current === false) {
      logging.navBarImpression({ home_icon_yn: withHomeButton ? "Y" : "N" });
      hasLogged.current = true;
    }
  }, [logging, withHomeButton]);
  return /* @__PURE__ */ (0, import_jsx_runtime6.jsx)(import_jsx_runtime6.Fragment, { children });
}

// src/components/NavigationBar/RNNavigationBar/Default.tsx
var import_jsx_runtime7 = require("react/jsx-runtime");
function DefaultNavigationBar() {
  const globals = getAppsInTossGlobals();
  const { open: openMoreButtonBottomSheet } = useMoreButtonBottomSheet();
  const parsedNavigationBar = globals.navigationBar != null ? safeParseNavigationBar(globals.navigationBar) : null;
  const withHomeButton = parsedNavigationBar?.withHomeButton ?? false;
  const withBackButton = parsedNavigationBar?.withBackButton ?? true;
  const initialAccessoryButton = parsedNavigationBar?.initialAccessoryButton;
  const navigationEvent = useNavigationEvent();
  useHardwareBackPress(navigationEvent.handleBack);
  return /* @__PURE__ */ (0, import_jsx_runtime7.jsx)(NavigationBarImpressionArea, { withHomeButton, children: /* @__PURE__ */ (0, import_jsx_runtime7.jsx)(
    import_private5.TopNavigation,
    {
      title: globals.brandDisplayName,
      icon: toIcon(globals.brandIcon),
      onPressDots: openMoreButtonBottomSheet,
      contentVisible: true,
      onPressTitle: withHomeButton ? navigationEvent.handleHomeButtonClick : void 0,
      onPressClose: navigationEvent.handleCloseButtonClick,
      withHome: withHomeButton,
      fixedRightButton: initialAccessoryButton,
      children: /* @__PURE__ */ (0, import_jsx_runtime7.jsx)(import_private5.NavigationLeft, { visible: withBackButton, children: /* @__PURE__ */ (0, import_jsx_runtime7.jsx)(import_private5.NavigationBackButton, { onPress: navigationEvent.handleBack, canGoBack: false }) })
    }
  ) });
}

// src/components/NavigationBar/RNNavigationBar/Game.tsx
var import_tds_react_native9 = require("@toss/tds-react-native");
var import_private6 = require("@toss/tds-react-native/private");
var import_react_native18 = require("react-native");
var import_jsx_runtime8 = require("react/jsx-runtime");
function GameNavigationBar() {
  const safeAreaTop = (0, import_private6.useSafeAreaTop)();
  const global2 = getAppsInTossGlobals();
  const { open: openMoreButtonBottomSheet } = useMoreButtonBottomSheet();
  const navigationEvent = useNavigationEvent();
  useHardwareBackPress(navigationEvent.handleBack);
  const parsedNavigationBar = global2.navigationBar != null ? safeParseNavigationBar(global2.navigationBar) : null;
  const initialAccessoryButton = parsedNavigationBar?.initialAccessoryButton;
  return /* @__PURE__ */ (0, import_jsx_runtime8.jsxs)(import_jsx_runtime8.Fragment, { children: [
    /* @__PURE__ */ (0, import_jsx_runtime8.jsx)(import_tds_react_native9.PageNavbar, { preference: { type: "none" } }),
    /* @__PURE__ */ (0, import_jsx_runtime8.jsx)(
      import_react_native18.View,
      {
        style: {
          width: "100%",
          // TODO: UI관련 스타일 설정은 TDS로 이관
          height: import_react_native18.Platform.OS === "ios" ? 44 : 54,
          flexDirection: "row",
          alignItems: "center",
          justifyContent: "flex-end",
          position: "absolute",
          zIndex: Z_INDEX.CLOSE_BUTTON,
          marginTop: safeAreaTop,
          paddingRight: 10
        },
        pointerEvents: "box-none",
        children: /* @__PURE__ */ (0, import_jsx_runtime8.jsx)(
          import_private6.NavigationRightContent,
          {
            fixedRightButton: initialAccessoryButton,
            onPressDots: openMoreButtonBottomSheet,
            onPressClose: navigationEvent.handleCloseButtonClick,
            theme: "dark"
          }
        )
      }
    )
  ] });
}

// src/components/NavigationBar/RNNavigationBar/index.tsx
var RNNavigationBar = {
  Default: DefaultNavigationBar,
  Game: GameNavigationBar
};

// src/bridge-entry.ts
var bridge_entry_exports = {};
__reExport(bridge_entry_exports, require("@apps-in-toss/native-modules/async-bridges"));
__reExport(bridge_entry_exports, require("@apps-in-toss/native-modules/constant-bridges"));
__reExport(bridge_entry_exports, require("@apps-in-toss/native-modules/event-bridges"));

// src/components/RNAppContainer.tsx
var import_jsx_runtime9 = require("react/jsx-runtime");
function RNAppContainer({ children }) {
  const global2 = getAppsInTossGlobals();
  switch (global2.appType) {
    case "game":
      return /* @__PURE__ */ (0, import_jsx_runtime9.jsx)(GameAppContainer, { children });
    case "general":
    default:
      return /* @__PURE__ */ (0, import_jsx_runtime9.jsx)(GeneralAppContainer, { children });
  }
}
function GameAppContainer({ children }) {
  const [isEntryMessageExited, setIsEntryMessageExited] = (0, import_react14.useState)(false);
  (0, import_react14.useEffect)(() => {
    if (import_react_native19.Platform.OS === "ios") {
      (0, import_native_modules12.setIosSwipeGestureEnabled)({ isEnabled: false });
      return () => {
        (0, import_native_modules12.setIosSwipeGestureEnabled)({ isEnabled: true });
      };
    }
    return;
  }, []);
  (0, import_react14.useEffect)(() => {
    import_native_modules12.appsInTossEvent.addEventListener("entryMessageExited", {
      onEvent: () => {
        setIsEntryMessageExited(true);
      }
    });
  }, []);
  return /* @__PURE__ */ (0, import_jsx_runtime9.jsxs)(import_jsx_runtime9.Fragment, { children: [
    /* @__PURE__ */ (0, import_jsx_runtime9.jsx)(RNNavigationBar.Game, {}),
    (0, bridge_entry_exports.getOperationalEnvironment)() === "toss" ? /* @__PURE__ */ (0, import_jsx_runtime9.jsx)(GameProfile, { isReadyForProfileUI: isEntryMessageExited, children }) : children
  ] });
}
function GeneralAppContainer({ children }) {
  return /* @__PURE__ */ (0, import_jsx_runtime9.jsxs)(import_jsx_runtime9.Fragment, { children: [
    /* @__PURE__ */ (0, import_jsx_runtime9.jsx)(RNNavigationBar.Default, {}),
    children
  ] });
}

// src/core/registerApp.tsx
var import_jsx_runtime10 = require("react/jsx-runtime");
function AppsInTossContainer(Container, { children, ...initialProps }) {
  if (!(0, import_native_modules13.isMinVersionSupported)({
    android: "5.220.0",
    ios: "5.221.0"
  })) {
    return /* @__PURE__ */ (0, import_jsx_runtime10.jsxs)(import_jsx_runtime10.Fragment, { children: [
      /* @__PURE__ */ (0, import_jsx_runtime10.jsx)(AppEvent.Entry, {}),
      /* @__PURE__ */ (0, import_jsx_runtime10.jsx)(AppEvent.System, { ...initialProps }),
      /* @__PURE__ */ (0, import_jsx_runtime10.jsx)(AppUpdate, {})
    ] });
  }
  return /* @__PURE__ */ (0, import_jsx_runtime10.jsxs)(import_jsx_runtime10.Fragment, { children: [
    /* @__PURE__ */ (0, import_jsx_runtime10.jsx)(AppEvent.StayTime, {}),
    /* @__PURE__ */ (0, import_jsx_runtime10.jsx)(AppEvent.Entry, {}),
    /* @__PURE__ */ (0, import_jsx_runtime10.jsx)(AppEvent.System, { ...initialProps }),
    /* @__PURE__ */ (0, import_jsx_runtime10.jsx)(Container, { ...initialProps, children: /* @__PURE__ */ (0, import_jsx_runtime10.jsx)(import_tds_react_native10.TDSProvider, { colorPreference: "light", token: { color: { primary: getAppsInTossGlobals().brandPrimaryColor } }, children: /* @__PURE__ */ (0, import_jsx_runtime10.jsx)(TDSContainer, { ...initialProps, children }) }) })
  ] });
}
function TDSContainer({ children }) {
  useAppsInTossBridge();
  return /* @__PURE__ */ (0, import_jsx_runtime10.jsx)(import_jsx_runtime10.Fragment, { children });
}
function registerApp(container, { context, analytics }) {
  const appName = getAppName();
  const isRegistered = import_react_native21.AppRegistry.getAppKeys().includes(appName);
  if (!isRegistered) {
    import_analytics.Analytics.init({
      logger: (params) => void (0, import_native_modules13.eventLog)(params),
      debug: analytics?.debug ?? __DEV__
    });
    const App = import_react_native20.Granite.registerApp(AppsInTossContainer.bind(null, container), {
      appName,
      context,
      setIosSwipeGestureEnabled: import_native_modules13.setIosSwipeGestureEnabled,
      router: {
        screenContainer: AppsInTossScreenContainer,
        defaultScreenOption: {
          statusBarStyle: "dark"
        }
      }
    });
    global.Page = App;
  }
  return global.Page;
}
function AppsInTossScreenContainer({ children }) {
  const isRNApp = getAppsInTossGlobals().webViewType == null;
  return /* @__PURE__ */ (0, import_jsx_runtime10.jsx)(import_analytics.Analytics.Screen, { children: isRNApp ? /* @__PURE__ */ (0, import_jsx_runtime10.jsx)(RNAppContainer, { children }) : children });
}
function getAppName() {
  try {
    return global.__granite.app.name;
  } catch (error) {
    console.error("unexpected error occurred while getting app name");
    throw error;
  }
}

// src/core/index.ts
var AppsInToss = {
  registerApp
};

// src/components/WebView.tsx
var import_native_modules18 = require("@apps-in-toss/native-modules");
var appsInTossAsyncBridges = __toESM(require("@apps-in-toss/native-modules/async-bridges"), 1);
var appsInTossConstantBridges = __toESM(require("@apps-in-toss/native-modules/constant-bridges"), 1);
var appsInTossEventBridges = __toESM(require("@apps-in-toss/native-modules/event-bridges"), 1);
var import_react_native31 = require("@granite-js/react-native");
var import_tds_react_native14 = require("@toss/tds-react-native");
var import_private9 = require("@toss/tds-react-native/private");
var import_react24 = require("react");
var import_react_native32 = require("react-native");

// src/components/GameWebView.tsx
var import_native_modules14 = require("@apps-in-toss/native-modules");
var import_react_native_webview = require("@granite-js/native/react-native-webview");
var import_react16 = require("react");
var import_react_native24 = require("react-native");

// src/components/NavigationBar/GameWebviewNavigationBar.tsx
var import_react_native22 = require("@granite-js/react-native");
var import_tds_react_native11 = require("@toss/tds-react-native");
var import_private7 = require("@toss/tds-react-native/private");
var import_es_hangul3 = require("es-hangul");
var import_react15 = require("react");
var import_react_native23 = require("react-native");
var import_jsx_runtime11 = require("react/jsx-runtime");
function GameWebviewNavigationBar() {
  const safeAreaTop = (0, import_private7.useSafeAreaTop)();
  const { openConfirm } = (0, import_tds_react_native11.useDialog)();
  const { captureExitLog } = useCaptureExitLog();
  const global2 = getAppsInTossGlobals();
  const logging = useNavigationBarLogging();
  const { open: openMoreButtonBottomSheet } = useMoreButtonBottomSheet();
  const parsedNavigationBar = global2.navigationBar != null ? safeParseNavigationBar(global2.navigationBar) : null;
  const initialAccessoryButton = parsedNavigationBar?.initialAccessoryButton;
  const handleGameWebviewClose = (0, import_react15.useCallback)(async () => {
    logging.closeButtonClick();
    const isConfirmed = await openConfirm({
      title: `${(0, import_es_hangul3.josa)(global2.brandDisplayName, "\uC744/\uB97C")} \uC885\uB8CC\uD560\uAE4C\uC694?`,
      leftButton: "\uCDE8\uC18C",
      rightButton: "\uC885\uB8CC\uD558\uAE30",
      closeOnDimmerClick: true,
      onEntered: logging.closePopupShow
    });
    logging.closePopupCtaClick(isConfirmed);
    if (isConfirmed) {
      captureExitLog(Date.now());
      (0, import_react_native22.closeView)();
    }
  }, [captureExitLog, global2.brandDisplayName, logging, openConfirm]);
  return /* @__PURE__ */ (0, import_jsx_runtime11.jsxs)(import_jsx_runtime11.Fragment, { children: [
    /* @__PURE__ */ (0, import_jsx_runtime11.jsx)(import_tds_react_native11.PageNavbar, { preference: { type: "none" } }),
    /* @__PURE__ */ (0, import_jsx_runtime11.jsx)(
      import_react_native23.View,
      {
        style: {
          width: "100%",
          // TODO: UI관련 스타일 설정은 TDS로 이관
          height: import_react_native23.Platform.OS === "ios" ? 44 : 54,
          flexDirection: "row",
          alignItems: "center",
          justifyContent: "flex-end",
          position: "absolute",
          zIndex: Z_INDEX.CLOSE_BUTTON,
          marginTop: safeAreaTop,
          paddingRight: 10
        },
        pointerEvents: "box-none",
        children: /* @__PURE__ */ (0, import_jsx_runtime11.jsx)(
          import_private7.NavigationRightContent,
          {
            fixedRightButton: initialAccessoryButton,
            onPressDots: openMoreButtonBottomSheet,
            onPressClose: () => {
              handleGameWebviewClose();
            },
            theme: "dark"
          }
        )
      }
    )
  ] });
}

// src/components/GameWebView.tsx
var import_jsx_runtime12 = require("react/jsx-runtime");
var GameWebView = (0, import_react16.forwardRef)(function GameWebView2(props, ref) {
  const [isEntryMessageExited, setIsEntryMessageExited] = (0, import_react16.useState)(false);
  (0, import_react16.useEffect)(() => {
    if (import_react_native24.Platform.OS === "ios") {
      (0, import_native_modules14.setIosSwipeGestureEnabled)({ isEnabled: false });
      return () => {
        (0, import_native_modules14.setIosSwipeGestureEnabled)({ isEnabled: true });
      };
    }
    return;
  }, []);
  (0, import_react16.useEffect)(() => {
    import_native_modules14.appsInTossEvent.addEventListener("entryMessageExited", {
      onEvent: () => {
        setIsEntryMessageExited(true);
      }
    });
  }, []);
  return /* @__PURE__ */ (0, import_jsx_runtime12.jsxs)(import_jsx_runtime12.Fragment, { children: [
    /* @__PURE__ */ (0, import_jsx_runtime12.jsx)(GameWebviewNavigationBar, {}),
    (0, import_native_modules14.getOperationalEnvironment)() === "toss" ? /* @__PURE__ */ (0, import_jsx_runtime12.jsx)(GameProfile, { isReadyForProfileUI: isEntryMessageExited, children: /* @__PURE__ */ (0, import_jsx_runtime12.jsx)(import_react_native_webview.WebView, { ref, ...props }) }) : /* @__PURE__ */ (0, import_jsx_runtime12.jsx)(import_react_native_webview.WebView, { ref, ...props })
  ] });
});

// src/components/PartnerWebView.tsx
var import_react_native_webview2 = require("@granite-js/native/react-native-webview");
var import_react18 = require("react");

// src/components/NavigationBar/PartnerWebviewNavigationBar.tsx
var import_react_native25 = require("@granite-js/react-native");
var import_tds_react_native12 = require("@toss/tds-react-native");
var import_private8 = require("@toss/tds-react-native/private");
var import_es_hangul4 = require("es-hangul");
var import_react17 = require("react");
var import_jsx_runtime13 = require("react/jsx-runtime");
function PartnerWebviewNavigationBar({ onBackButtonClick, onHomeButtonClick }) {
  const globals = getAppsInTossGlobals();
  const { captureExitLog } = useCaptureExitLog();
  const logging = useNavigationBarLogging();
  const { openConfirm } = (0, import_tds_react_native12.useDialog)();
  const { open: openMoreButtonBottomSheet } = useMoreButtonBottomSheet();
  const parsedNavigationBar = globals.navigationBar != null ? safeParseNavigationBar(globals.navigationBar) : null;
  const withHomeButton = parsedNavigationBar?.withHomeButton ?? false;
  const withBackButton = parsedNavigationBar?.withBackButton ?? true;
  const initialAccessoryButton = parsedNavigationBar?.initialAccessoryButton;
  const handleClose = (0, import_react17.useCallback)(async () => {
    logging.closeButtonClick();
    const isConfirmed = await openConfirm({
      title: `${(0, import_es_hangul4.josa)(globals.brandDisplayName, "\uC744/\uB97C")} \uC885\uB8CC\uD560\uAE4C\uC694?`,
      leftButton: "\uCDE8\uC18C",
      rightButton: "\uC885\uB8CC\uD558\uAE30",
      closeOnDimmerClick: true,
      onEntered: logging.closePopupShow
    });
    logging.closePopupCtaClick(isConfirmed);
    if (isConfirmed) {
      captureExitLog(Date.now());
      (0, import_react_native25.closeView)();
    }
  }, [captureExitLog, globals.brandDisplayName, logging, openConfirm]);
  return /* @__PURE__ */ (0, import_jsx_runtime13.jsx)(NavigationBarImpressionArea, { withHomeButton, children: /* @__PURE__ */ (0, import_jsx_runtime13.jsx)(
    import_private8.TopNavigation,
    {
      title: globals.brandDisplayName,
      icon: toIcon(globals.brandIcon),
      onPressDots: openMoreButtonBottomSheet,
      contentVisible: true,
      onPressTitle: withHomeButton ? onHomeButtonClick : void 0,
      onPressClose: handleClose,
      withHome: withHomeButton,
      fixedRightButton: initialAccessoryButton,
      children: /* @__PURE__ */ (0, import_jsx_runtime13.jsx)(import_private8.NavigationLeft, { visible: withBackButton, children: /* @__PURE__ */ (0, import_jsx_runtime13.jsx)(import_private8.NavigationBackButton, { onPress: onBackButtonClick, canGoBack: false }) })
    }
  ) });
}

// src/components/PartnerWebView.tsx
var import_jsx_runtime14 = require("react/jsx-runtime");
var PartnerWebView = (0, import_react18.forwardRef)(function PartnerWebViewScreen({ onBackButtonClick, onHomeButtonClick, ...webViewProps }, ref) {
  return /* @__PURE__ */ (0, import_jsx_runtime14.jsxs)(import_jsx_runtime14.Fragment, { children: [
    /* @__PURE__ */ (0, import_jsx_runtime14.jsx)(PartnerWebviewNavigationBar, { onBackButtonClick, onHomeButtonClick }),
    /* @__PURE__ */ (0, import_jsx_runtime14.jsx)(import_react_native_webview2.WebView, { ref, ...webViewProps, style: { flex: 1 } })
  ] });
});

// src/bridge-handler/useBridgeHandler.tsx
var import_react19 = require("react");
function serializeError(error) {
  return JSON.stringify(error, (_, value) => {
    if (value instanceof Error) {
      return {
        ...Object.entries(value).reduce((acc, [key, value2]) => {
          acc[key] = value2;
          return acc;
        }, {}),
        name: value.name,
        message: value.message,
        stack: value.stack,
        __isError: true
      };
    }
    return value;
  });
}
function methodHandler({
  args,
  eventId,
  functionName,
  handlerMap,
  injectJavaScript
}) {
  const func = handlerMap[functionName];
  if (!func) {
    console.error(`${functionName} is not a function`);
    return;
  }
  const wrappedFunc = async (...args2) => {
    const result = await func(...args2);
    return result;
  };
  wrappedFunc(...args).then((result) => {
    injectJavaScript?.(`
      window.__GRANITE_NATIVE_EMITTER.emit('${functionName}/resolve/${eventId}', ${JSON.stringify(result, null, 0)});
    `);
  }).catch((error) => {
    const serializedError = serializeError(error);
    injectJavaScript?.(`
      window.__GRANITE_NATIVE_EMITTER.emit('${functionName}/reject/${eventId}', ${serializedError});
    `);
  });
}
var globalEventListenerMap = /* @__PURE__ */ new Map();
function useBridgeHandler({
  onMessage,
  constantHandlerMap,
  asyncHandlerMap,
  eventListenerMap,
  injectedJavaScript: originalInjectedJavaScript
}) {
  const ref = (0, import_react19.useRef)(null);
  const injectedJavaScript = (0, import_react19.useMemo)(
    () => [
      `window.__CONSTANT_HANDLER_MAP = ${JSON.stringify(
        Object.entries(constantHandlerMap).reduce(
          (acc, [key, value]) => {
            acc[key] = typeof value === "function" ? value() : value;
            return acc;
          },
          {}
        )
      )};`,
      originalInjectedJavaScript,
      "true"
    ].join("\n"),
    [constantHandlerMap, originalInjectedJavaScript]
  );
  const createHandleOnEvent = (functionName, eventId) => (response) => {
    ref.current?.injectJavaScript(`
      window.__GRANITE_NATIVE_EMITTER.emit('${functionName}/onEvent/${eventId}', ${JSON.stringify(response, null, 0)});
    `);
  };
  const createHandleOnError = (functionName, eventId) => (error) => {
    const serializedError = serializeError(error);
    ref.current?.injectJavaScript(`
      window.__GRANITE_NATIVE_EMITTER.emit('${functionName}/onError/${eventId}', ${serializedError});
    `);
  };
  const $onMessage = (0, import_react19.useCallback)(
    async (e) => {
      onMessage?.(e);
      const data = JSON.parse(e.nativeEvent.data);
      if (typeof data !== "object" || data === null || typeof data.functionName !== "string" || typeof data.eventId !== "string" || typeof data.type !== "string" || !["addEventListener", "removeEventListener", "method"].includes(data.type)) {
        return;
      }
      switch (data.type) {
        case "addEventListener": {
          const handleOnEvent = createHandleOnEvent(data.functionName, data.eventId);
          const handleOnError = createHandleOnError(data.functionName, data.eventId);
          const remove = eventListenerMap[data.functionName]?.({
            onEvent: handleOnEvent,
            onError: handleOnError,
            options: data.args
          });
          if (remove) {
            globalEventListenerMap.set(`${data.functionName}/${data.eventId}`, remove);
          }
          break;
        }
        case "removeEventListener": {
          const key = `${data.functionName}/${data.eventId}`;
          const remove = globalEventListenerMap.get(key);
          remove?.();
          globalEventListenerMap.delete(key);
          break;
        }
        case "method": {
          methodHandler({
            args: data.args,
            eventId: data.eventId,
            functionName: data.functionName,
            handlerMap: asyncHandlerMap,
            injectJavaScript: ref.current?.injectJavaScript
          });
          break;
        }
      }
    },
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [onMessage]
  );
  return {
    ref,
    injectedJavaScript,
    onMessage: $onMessage
  };
}

// src/core/hooks/useWebBackHandler.tsx
var import_react_native26 = require("@granite-js/react-native");
var import_tds_react_native13 = require("@toss/tds-react-native");
var import_es_hangul5 = require("es-hangul");
var import_react21 = require("react");

// src/hooks/useWebviewHistoryStack.tsx
var import_react20 = require("react");
var INITIAL_STATE = { stack: [], index: -1 };
function reducer(state, action) {
  switch (action.type) {
    case "NAVIGATION_CHANGE": {
      const { url, canGoForward } = action;
      if (state.stack.length === 0) {
        return { stack: [url], index: 0 };
      }
      const { stack, index } = state;
      const cur = stack[index];
      if (url === cur) {
        return state;
      }
      const prev = index > 0 ? stack[index - 1] : void 0;
      const next = index < stack.length - 1 ? stack[index + 1] : void 0;
      if (prev && url === prev && canGoForward) {
        return { ...state, index: index - 1 };
      }
      if (next && url === next) {
        return { ...state, index: index + 1 };
      }
      const base = stack.slice(0, index + 1);
      const nextStack = [...base, url];
      return { stack: nextStack, index: nextStack.length - 1 };
    }
    default:
      return state;
  }
}
function useWebViewHistory() {
  const [state, dispatch] = (0, import_react20.useReducer)(reducer, INITIAL_STATE);
  const onNavigationStateChange = (0, import_react20.useCallback)(({ url, canGoForward: canGoForward2 }) => {
    dispatch({ type: "NAVIGATION_CHANGE", url, canGoForward: canGoForward2 });
  }, []);
  const { canGoBack, canGoForward } = (0, import_react20.useMemo)(() => {
    const canBack = state.index > 0;
    const canFwd = state.index >= 0 && state.index < state.stack.length - 1;
    return { canGoBack: canBack, canGoForward: canFwd };
  }, [state.index, state.stack.length]);
  return { onNavigationStateChange, hasHistory: canGoBack, canGoForward };
}

// src/core/hooks/useWebBackHandler.tsx
var HISTORY_BACK_SCRIPT = `
(function() {
  window.history.back();
  true;
})();
`;
var HISTORY_HOME_SCRIPT = `
  (function() {
    window.location.href = '/';
    true;
  })();
`;
function useWebBackHandler(webViewRef) {
  const { captureExitLog } = useCaptureExitLog();
  const { hasHistory, onNavigationStateChange } = useWebViewHistory();
  const {
    handlersRef: webBackHandlersRef,
    hasBackEvent: hasWebBackEvent,
    addEventListener: addWebBackEventListener,
    removeEventListener: removeWebBackEventListener
  } = (0, import_react_native26.useBackEventState)();
  const logging = useNavigationBarLogging();
  const { openConfirm } = (0, import_tds_react_native13.useDialog)();
  const global2 = getAppsInTossGlobals();
  const addEventListener = (0, import_react21.useCallback)(
    (handler) => {
      addWebBackEventListener(handler);
    },
    [addWebBackEventListener]
  );
  const removeEventListener = (0, import_react21.useCallback)(
    (handler) => {
      removeWebBackEventListener(handler);
    },
    [removeWebBackEventListener]
  );
  const handleWebBack = (0, import_react21.useCallback)(async () => {
    if (hasWebBackEvent) {
      for (const handler of webBackHandlersRef) {
        handler();
      }
      return;
    }
    if (hasHistory) {
      webViewRef.current?.injectJavaScript(HISTORY_BACK_SCRIPT);
    } else {
      const isConfirmed = await openConfirm({
        title: `${(0, import_es_hangul5.josa)(global2.brandDisplayName, "\uC744/\uB97C")} \uC885\uB8CC\uD560\uAE4C\uC694?`,
        leftButton: "\uCDE8\uC18C",
        rightButton: "\uC885\uB8CC\uD558\uAE30",
        closeOnDimmerClick: true,
        onEntered: logging.closePopupShow
      });
      logging.closePopupCtaClick(isConfirmed);
      if (isConfirmed) {
        captureExitLog(Date.now());
        (0, import_react_native26.closeView)();
      }
    }
  }, [
    captureExitLog,
    global2.brandDisplayName,
    hasHistory,
    hasWebBackEvent,
    webBackHandlersRef,
    logging,
    openConfirm,
    webViewRef
  ]);
  const handleWebHome = (0, import_react21.useCallback)(() => {
    logging.homeButtonClick();
    if (hasWebBackEvent) {
      for (const handler of webBackHandlersRef) {
        handler();
      }
      return;
    }
    webViewRef.current?.injectJavaScript(HISTORY_HOME_SCRIPT);
  }, [hasWebBackEvent, webBackHandlersRef, logging, webViewRef]);
  return (0, import_react21.useMemo)(
    () => ({ addEventListener, removeEventListener, handleWebBack, handleWebHome, onNavigationStateChange }),
    [addEventListener, removeEventListener, handleWebBack, handleWebHome, onNavigationStateChange]
  );
}

// src/core/utils/mergeRefs.ts
function mergeRefs(...refs) {
  return (value) => {
    refs.forEach((ref) => {
      if (typeof ref === "function") {
        ref(value);
      } else if (ref != null) {
        ref.current = value;
      }
    });
  };
}

// src/hooks/useCreateUserAgent.ts
var import_native_modules15 = require("@apps-in-toss/native-modules");
var import_react_native27 = require("react-native");
var FontA11yCategory = {
  Large: "Large",
  xLarge: "xLarge",
  xxLarge: "xxLarge",
  xxxLarge: "xxxLarge",
  A11y_Medium: "A11y_Medium",
  A11y_Large: "A11y_Large",
  A11y_xLarge: "A11y_xLarge",
  A11y_xxLarge: "A11y_xxLarge",
  A11y_xxxLarge: "A11y_xxxLarge"
};
var androidFontScaleMap = {
  100: FontA11yCategory.Large,
  110: FontA11yCategory.xLarge,
  120: FontA11yCategory.xxLarge,
  135: FontA11yCategory.xxxLarge,
  160: FontA11yCategory.A11y_Medium,
  190: FontA11yCategory.A11y_Large,
  235: FontA11yCategory.A11y_xLarge,
  275: FontA11yCategory.A11y_xxLarge,
  310: FontA11yCategory.A11y_xxxLarge
};
var iosScaleToAndroidScale = {
  0.823: 100,
  0.882: 100,
  0.941: 100,
  1: 100,
  1.118: 110,
  1.235: 120,
  1.353: 135,
  1.786: 160,
  2.143: 190,
  2.643: 235,
  3.143: 275,
  3.571: 310
};
function convertToAndroidStyleScale(fontScale, platform) {
  if (platform === "android") {
    if (fontScale <= 1) {
      return 100;
    }
    const scaledValue = Math.round(fontScale * 100);
    const keys = Object.keys(androidFontScaleMap).map(Number).sort((a, b) => a - b);
    let closestKey = keys[0];
    let minDiff = Math.abs(scaledValue - closestKey);
    for (const key of keys) {
      const diff = Math.abs(scaledValue - key);
      if (diff < minDiff) {
        minDiff = diff;
        closestKey = key;
      }
    }
    return closestKey;
  } else {
    const iosScales = Object.keys(iosScaleToAndroidScale).map(Number);
    let closestScale = iosScales[0];
    let minDiff = Math.abs(fontScale - closestScale);
    for (const scale of iosScales) {
      const diff = Math.abs(fontScale - scale);
      if (diff < minDiff) {
        minDiff = diff;
        closestScale = scale;
      }
    }
    return iosScaleToAndroidScale[closestScale];
  }
}
function mapIOSFontScaleToCategory(fontScale) {
  if (fontScale < 1) {
    return FontA11yCategory.Large;
  }
  if (Math.abs(fontScale - 1) < 0.05) {
    return FontA11yCategory.Large;
  }
  if (Math.abs(fontScale - 1.118) < 0.05) {
    return FontA11yCategory.xLarge;
  }
  if (Math.abs(fontScale - 1.235) < 0.05) {
    return FontA11yCategory.xxLarge;
  }
  if (Math.abs(fontScale - 1.353) < 0.05) {
    return FontA11yCategory.xxxLarge;
  }
  if (Math.abs(fontScale - 1.786) < 0.05) {
    return FontA11yCategory.A11y_Medium;
  }
  if (Math.abs(fontScale - 2.143) < 0.05) {
    return FontA11yCategory.A11y_Large;
  }
  if (Math.abs(fontScale - 2.643) < 0.05) {
    return FontA11yCategory.A11y_xLarge;
  }
  if (Math.abs(fontScale - 3.143) < 0.05) {
    return FontA11yCategory.A11y_xxLarge;
  }
  if (Math.abs(fontScale - 3.571) < 0.05) {
    return FontA11yCategory.A11y_xxxLarge;
  }
  return FontA11yCategory.Large;
}
function mapAndroidFontScaleToCategory(fontScale) {
  if (fontScale <= 1) {
    return androidFontScaleMap[100];
  }
  const scaledValue = Math.round(fontScale * 100);
  const keys = Object.keys(androidFontScaleMap).map(Number).sort((a, b) => a - b);
  if (keys.length === 0) {
    return androidFontScaleMap[100];
  }
  let closestKey = keys[0];
  let minDiff = Math.abs(scaledValue - closestKey);
  for (const key of keys) {
    const diff = Math.abs(scaledValue - key);
    if (diff < minDiff) {
      minDiff = diff;
      closestKey = key;
    }
  }
  return androidFontScaleMap[closestKey];
}
function mapFontScaleToCategory(fontScale, platform) {
  return platform === "ios" ? mapIOSFontScaleToCategory(fontScale) : mapAndroidFontScaleToCategory(fontScale);
}
function useCreateUserAgent({
  batteryModePreference,
  colorPreference,
  locale,
  navbarPreference,
  pureSafeArea,
  safeArea,
  safeAreaBottomTransparency
}) {
  const platform = (0, import_native_modules15.getPlatformOS)();
  const appVersion = (0, import_native_modules15.getTossAppVersion)();
  const { fontScale } = (0, import_react_native27.useWindowDimensions)();
  const platformString = platform === "ios" ? "iPhone" : "Android";
  const fontA11y = mapFontScaleToCategory(fontScale, platform);
  const normalizedFontScale = convertToAndroidStyleScale(fontScale, platform);
  return [
    `TossApp/${appVersion}`,
    batteryModePreference && `TossBatteryModePreference/${batteryModePreference}`,
    colorPreference && `TossColorPreference/${colorPreference}`,
    `TossFontAccessibility/${fontA11y}`,
    `TossFontScale/${normalizedFontScale}`,
    locale && `TossLocale/${locale}`,
    navbarPreference && `TossNavbarPreference/${navbarPreference}`,
    pureSafeArea && `TossPureSafeArea/${pureSafeArea}`,
    safeArea && `TossSafeArea/${safeArea}`,
    safeAreaBottomTransparency && `TossSafeAreaBottomTransparency/${safeAreaBottomTransparency}`,
    platformString
  ].filter(Boolean).join(" ");
}

// src/hooks/useGeolocation.ts
var import_native_modules16 = require("@apps-in-toss/native-modules");
var import_react_native28 = require("@granite-js/react-native");
var import_react22 = require("react");
function useGeolocation({ accuracy, distanceInterval, timeInterval }) {
  const isVisible = (0, import_react_native28.useVisibility)();
  const [location, setLocation] = (0, import_react22.useState)(null);
  (0, import_react22.useEffect)(() => {
    if (!isVisible) {
      return;
    }
    return (0, import_native_modules16.startUpdateLocation)({
      options: {
        accuracy,
        distanceInterval,
        timeInterval
      },
      onEvent: setLocation,
      onError: console.error
    });
  }, [accuracy, distanceInterval, timeInterval, isVisible]);
  return location;
}

// src/hooks/useWaitForReturnNavigator.tsx
var import_react_native29 = require("@granite-js/react-native");
var import_react23 = require("react");
function useWaitForReturnNavigator() {
  const callbacks = (0, import_react23.useRef)([]).current;
  const navigation = (0, import_react_native29.useNavigation)();
  const startNavigating = (0, import_react23.useCallback)(
    (route, params) => {
      return new Promise((resolve) => {
        callbacks.push(resolve);
        navigation.navigate(route, params);
      });
    },
    [callbacks, navigation]
  );
  const handleVisibilityChange = (0, import_react23.useCallback)(
    (state) => {
      if (state === "visible" && callbacks.length > 0) {
        for (const callback of callbacks) {
          callback();
        }
        callbacks.splice(0, callbacks.length);
      }
    },
    [callbacks]
  );
  (0, import_react_native29.useVisibilityChange)(handleVisibilityChange);
  return startNavigating;
}

// src/utils/log.ts
var import_native_modules17 = require("@apps-in-toss/native-modules");
var import_react_native30 = require("@granite-js/react-native");

// src/utils/extractDateFromUUIDv7.ts
var extractDateFromUUIDv7 = (uuid) => {
  const timestampHex = uuid.split("-").join("").slice(0, 12);
  const timestamp = Number.parseInt(timestampHex, 16);
  return new Date(timestamp);
};

// src/utils/log.ts
var getGroupId = (url) => {
  try {
    const urlObject = new URL(url);
    return {
      groupId: urlObject.pathname,
      search: urlObject.search.startsWith("?") ? urlObject.search.substring(1) : urlObject.search
    };
  } catch {
    return {
      groupId: "unknown",
      search: "unknown"
    };
  }
};
var getReferrer = () => {
  try {
    const referrer = new URL((0, import_react_native30.getSchemeUri)());
    return referrer.searchParams.get("referrer");
  } catch {
    return "";
  }
};
var trackScreen = (url) => {
  const { groupId, search } = getGroupId(url);
  const log = {
    log_type: "screen",
    log_name: `${groupId}::screen`,
    params: {
      search,
      referrer: getReferrer(),
      deployment_id: env.getDeploymentId(),
      deployment_timestamp: extractDateFromUUIDv7(env.getDeploymentId()).getTime()
    }
  };
  return (0, import_native_modules17.eventLog)(log);
};

// src/components/WebView.tsx
var import_jsx_runtime15 = require("react/jsx-runtime");
var operationalEnvironment = appsInTossConstantBridges.getOperationalEnvironment();
var TYPES = ["partner", "external", "game"];
var WEBVIEW_TYPES = {
  partner: PartnerWebView,
  external: import_tds_react_native14.ExternalWebViewScreen,
  game: GameWebView
};
function mergeSchemeQueryParamsInto(url) {
  const baseUrl = new URL(url);
  const schemeUrl = new URL((0, import_react_native31.getSchemeUri)());
  baseUrl.pathname = schemeUrl.pathname;
  for (const [key, value] of schemeUrl.searchParams.entries()) {
    baseUrl.searchParams.set(key, value);
  }
  return baseUrl;
}
function getWebViewUri(local) {
  if (__DEV__) {
    const devUrl = `http://${local.host}:${local.port}`;
    return mergeSchemeQueryParamsInto(devUrl).toString();
  }
  const { url: rawUrl } = import_native_modules18.AppsInTossModule.getWebBundleURL({});
  const url = mergeSchemeQueryParamsInto(rawUrl);
  const deploymentId = env.getDeploymentId();
  if (deploymentId) {
    url.searchParams.set("_deploymentId", deploymentId);
  }
  return url.toString();
}
function WebView({ type, local, onMessage, ...props }) {
  if (!TYPES.includes(type)) {
    throw new Error(`Invalid WebView type: '${type}'`);
  }
  const webViewRef = (0, import_react24.useRef)(null);
  const webBackHandler = useWebBackHandler(webViewRef);
  const uri = (0, import_react24.useMemo)(() => getWebViewUri(local), [local]);
  const top = (0, import_private9.useSafeAreaTop)();
  const bottom = (0, import_private9.useSafeAreaBottom)();
  const global2 = getAppsInTossGlobals();
  const topNavigation = (0, import_private9.useTopNavigation)();
  const disableTextSelectionCSS = `
    (function() {
      const style = document.createElement('style');
      style.textContent = '*:not(input):not(textarea) { -webkit-user-select: none; -moz-user-select: none; -ms-user-select: none; user-select: none; -webkit-touch-callout: none; }';
      document.head.appendChild(style);
    })();
  `;
  const [allowsBackForwardNavigationGestures, setAllowsBackForwardNavigationGestures] = (0, import_react24.useState)(
    props.allowsBackForwardNavigationGestures
  );
  const handler = useBridgeHandler({
    onMessage,
    injectedJavaScript: [disableTextSelectionCSS].join("\n"),
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    eventListenerMap: {
      ...appsInTossEventBridges,
      navigationAccessoryEvent: ({ onEvent, onError }) => import_tds_react_native14.tdsEvent.addEventListener("navigationAccessoryEvent", { onEvent, onError }),
      backEvent: ({ onEvent }) => {
        webBackHandler.addEventListener(onEvent);
        return () => {
          webBackHandler.removeEventListener(onEvent);
        };
      },
      entryMessageExited: ({ onEvent, onError }) => import_native_modules18.appsInTossEvent.addEventListener("entryMessageExited", { onEvent, onError }),
      updateLocationEvent: ({ onEvent, onError, options }) => import_native_modules18.appsInTossEvent.addEventListener("updateLocationEvent", { onEvent, onError, options }),
      /** @internal */
      appBridgeCallbackEvent: ({ onEvent, onError, options }) => import_native_modules18.appsInTossEvent.addEventListener("appBridgeCallbackEvent", { onEvent, onError, options }),
      /** AdMob */
      loadAdMobInterstitialAd: import_native_modules18.GoogleAdMob.loadAdMobInterstitialAd,
      showAdMobInterstitialAd: import_native_modules18.GoogleAdMob.showAdMobInterstitialAd,
      loadAdMobRewardedAd: import_native_modules18.GoogleAdMob.loadAdMobRewardedAd,
      showAdMobRewardedAd: import_native_modules18.GoogleAdMob.showAdMobRewardedAd,
      /** AdMobV2 */
      loadAppsInTossAdMob: import_native_modules18.GoogleAdMob.loadAppsInTossAdMob,
      showAppsInTossAdMob: import_native_modules18.GoogleAdMob.showAppsInTossAdMob,
      /** IAP */
      iapCreateOneTimePurchaseOrder: import_native_modules18.IAP.createOneTimePurchaseOrder,
      requestOneTimePurchase: import_native_modules18.requestOneTimePurchase
    },
    constantHandlerMap: {
      ...appsInTossConstantBridges,
      getSafeAreaTop: () => top,
      getSafeAreaBottom: () => bottom,
      ...Object.fromEntries(Object.entries(global2).map(([key, value]) => [key, () => value])),
      /** AdMob */
      loadAdMobInterstitialAd_isSupported: import_native_modules18.GoogleAdMob.loadAdMobInterstitialAd.isSupported,
      showAdMobInterstitialAd_isSupported: import_native_modules18.GoogleAdMob.showAdMobInterstitialAd.isSupported,
      loadAdMobRewardedAd_isSupported: import_native_modules18.GoogleAdMob.loadAdMobRewardedAd.isSupported,
      showAdMobRewardedAd_isSupported: import_native_modules18.GoogleAdMob.showAdMobRewardedAd.isSupported,
      /** AdMobV2 */
      loadAppsInTossAdMob_isSupported: import_native_modules18.GoogleAdMob.loadAppsInTossAdMob.isSupported,
      showAppsInTossAdMob_isSupported: import_native_modules18.GoogleAdMob.showAppsInTossAdMob.isSupported,
      /** env */
      getDeploymentId: env.getDeploymentId
    },
    asyncHandlerMap: {
      ...appsInTossAsyncBridges,
      setIosSwipeGestureEnabled: (options) => {
        setAllowsBackForwardNavigationGestures(options.isEnabled);
        return appsInTossAsyncBridges.setIosSwipeGestureEnabled(options);
      },
      addAccessoryButton: async (params) => topNavigation.addAccessoryButton(params),
      removeAccessoryButton: async () => topNavigation.removeAccessoryButton(),
      /** permissions */
      requestPermission: appsInTossAsyncBridges.requestPermission,
      openPermissionDialog: appsInTossAsyncBridges.openPermissionDialog,
      getPermission: appsInTossAsyncBridges.getPermission,
      getClipboardText: appsInTossAsyncBridges.getClipboardText,
      setClipboardText: appsInTossAsyncBridges.setClipboardText,
      fetchAlbumPhotos: appsInTossAsyncBridges.fetchAlbumPhotos,
      fetchContacts: appsInTossAsyncBridges.fetchContacts,
      getCurrentLocation: appsInTossAsyncBridges.getCurrentLocation,
      openCamera: appsInTossAsyncBridges.openCamera,
      /** Storage */
      getStorageItem: import_native_modules18.Storage.getItem,
      setStorageItem: import_native_modules18.Storage.setItem,
      removeStorageItem: import_native_modules18.Storage.removeItem,
      clearItems: import_native_modules18.Storage.clearItems,
      /** IAP */
      iapGetProductItemList: import_native_modules18.IAP.getProductItemList,
      iapCreateOneTimePurchaseOrder: import_native_modules18.iapCreateOneTimePurchaseOrder,
      processProductGrant: import_native_modules18.processProductGrant,
      getPendingOrders: import_native_modules18.IAP.getPendingOrders,
      getCompletedOrRefundedOrders: import_native_modules18.IAP.getCompletedOrRefundedOrders,
      completeProductGrant: import_native_modules18.IAP.completeProductGrant
    }
  });
  const headerPropForExternalWebView = (0, import_react24.useMemo)(() => {
    const parsedNavigationBar = global2.navigationBar != null ? safeParseNavigationBar(global2.navigationBar) : null;
    const initialAccessoryButton = parsedNavigationBar?.initialAccessoryButton;
    const withBackButton = parsedNavigationBar?.withBackButton ?? true;
    if (type === "external" && initialAccessoryButton != null) {
      return {
        header: {
          withBackButton,
          title: initialAccessoryButton.title,
          icon: initialAccessoryButton.icon
        }
      };
    }
    return {};
  }, [global2.navigationBar, type]);
  const BaseWebView = WEBVIEW_TYPES[type];
  const webViewDebuggingEnabled = operationalEnvironment === "sandbox";
  const userAgent = useCreateUserAgent({
    colorPreference: "light"
  });
  const refs = mergeRefs(handler.ref, webViewRef);
  (0, import_react24.useEffect)(() => {
    const callback = () => {
      webBackHandler.handleWebBack();
      return true;
    };
    import_react_native32.BackHandler.addEventListener("hardwareBackPress", callback);
    return () => import_react_native32.BackHandler.removeEventListener("hardwareBackPress", callback);
  }, [webBackHandler]);
  return /* @__PURE__ */ (0, import_jsx_runtime15.jsx)(
    BaseWebView,
    {
      ref: refs,
      ...props,
      ...headerPropForExternalWebView,
      source: {
        uri,
        // NOTE: https://github.com/react-native-webview/react-native-webview/pull/3133
        headers: {
          "User-Agent": userAgent
        }
      },
      onHomeButtonClick: webBackHandler.handleWebHome,
      onBackButtonClick: webBackHandler.handleWebBack,
      onNavigationStateChange: (event) => {
        if (event.url) {
          trackScreen(event.url);
        }
        props.onNavigationStateChange?.(event);
        webBackHandler.onNavigationStateChange(event);
      },
      userAgent: import_react_native32.Platform.OS === "ios" ? userAgent : void 0,
      sharedCookiesEnabled: true,
      webviewDebuggingEnabled: webViewDebuggingEnabled,
      thirdPartyCookiesEnabled: true,
      onMessage: handler.onMessage,
      injectedJavaScript: handler.injectedJavaScript,
      injectedJavaScriptBeforeContentLoaded: handler.injectedJavaScript,
      decelerationRate: import_react_native32.Platform.OS === "ios" ? 1 : void 0,
      allowsBackForwardNavigationGestures
    }
  );
}

// src/index.ts
__reExport(src_exports, require("@apps-in-toss/analytics"), module.exports);
var import_private10 = require("@toss/tds-react-native/private");
__reExport(src_exports, require("@apps-in-toss/native-modules"), module.exports);
__reExport(src_exports, require("@apps-in-toss/types"), module.exports);
var Analytics2 = {
  init: import_analytics2.Analytics.init,
  Impression: import_analytics2.Analytics.Impression,
  Press: import_analytics2.Analytics.Press,
  Area: import_analytics2.Analytics.Area
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Analytics,
  AppsInToss,
  INTERNAL__onVisibilityChangedByTransparentServiceWeb,
  OverlayProvider,
  WebView,
  env,
  useCreateUserAgent,
  useGeolocation,
  useOverlay,
  useTopNavigation,
  useWaitForReturnNavigator,
  ...require("@apps-in-toss/analytics"),
  ...require("@apps-in-toss/native-modules"),
  ...require("@apps-in-toss/types")
});
